crates/bevy_render/src/texture/image.rs

```diff
@@ -99,6 +108,29 @@ impl ImageFormat {
             ImageFormat::Basis | ImageFormat::Ktx2 => return None,
         })
     }
+
+    /// Attempts to convert from [`image::ImageFormat`].
+    ///
+    /// Unsupported formats will become [`None`].
+    pub fn from_image_crate_format(format: image::ImageFormat) -> Option<Self> {
```

Comment:
Minor nit, could this be replaced with an implementation of `TryFrom` instead?

<details>
  <summary>Example</summary>
  
```rust
/// Error returned when converting between Bevy and [image] image format types.
#[derive(Debug, Error)]
#[error("Image format '{image:?}' maps to '{bevy:?}'")]
pub struct UnsupportedImageFormatMapping {
    /// Image format type from [image]
    pub image: Option<image::ImageFormat>,
    /// Bevy's image format type
    pub bevy: Option<ImageFormat>,
}

impl TryFrom<image::ImageFormat> for ImageFormat {
    type Error = UnsupportedImageFormatMapping;

    fn try_from(format: image::ImageFormat) -> Result<Self, Self::Error> {
        let format = match format {
            image::ImageFormat::Avif => ImageFormat::Avif,
            image::ImageFormat::Bmp => ImageFormat::Bmp,
            // ...
            _ => {
                return Err(Self::Error {
                    image: Some(format),
                    bevy: None,
                })
            }
        };

        Ok(format)
    }
}

impl TryFrom<ImageFormat> for image::ImageFormat {
    type Error = UnsupportedImageFormatMapping;

    fn try_from(format: ImageFormat) -> Result<Self, Self::Error> {
        let format = match format {
            ImageFormat::Avif => image::ImageFormat::Avif,
            ImageFormat::Bmp => image::ImageFormat::Bmp,
            // ...
            _ => {
                return Err(Self::Error {
                    image: None,
                    bevy: Some(format),
                })
            }
        };

        Ok(format)
    }
}

impl ImageFormat {
    // ...

    /// Attempts to convert to [`image::ImageFormat`].
    ///
    /// Unsupported formats will become [`None`].
    pub fn as_image_crate_format(&self) -> Option<image::ImageFormat> {
        (*self).try_into().ok()
    }

    /// Attempts to convert from [`image::ImageFormat`].
    ///
    /// Unsupported formats will become [`None`].
    pub fn from_image_crate_format(format: image::ImageFormat) -> Option<Self> {
        format.try_into().ok()
    }

    // ...
}
```
  
</details>

I personally think it's nicer to use the core traits when appropriate.